File: treeView/src/main/java/org/sqlunet/treeview/control/TextController.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.sqlunet.treeview.control;

import android.content.Context;
import android.view.View;
import android.widget.TextView;

import org.sqlunet.treeview.model.TreeNode;

import androidx.annotation.NonNull;

/**
 * Text controller
 *
 * @author Bogdan Melnychuk on 2/11/15.
 */
public class TextController extends Controller<Object>
{
	/**
	 * Constructor
	 *
	 * @param breakExpand whether this controller breaks expansion
	 */
	public TextController(final boolean breakExpand)
	{
		super(breakExpand);
	}

	@NonNull
	@Override
	public View createNodeView(@NonNull final Context context, final TreeNode node, final Object value, final int minHeight)
	{
		final TextView textView = new TextView(context);
		if (minHeight > 0)
		{
			textView.setMinimumHeight(minHeight);
		}

		if (value instanceof CompositeValue)
		{
			final CompositeValue data = (CompositeValue) value;
			textView.setText(data.text);
			textView.setCompoundDrawablePadding(10);
			textView.setCompoundDrawablesWithIntrinsicBounds(data.icon, 0, 0, 0);
		}
		else if (value instanceof CharSequence)
		{
			textView.setText((CharSequence) value);
		}
		else
		{
			throw new IllegalArgumentException(value.toString());
		}
		return textView;
	}
}

File: treeView/src/main/java/org/sqlunet/treeview/control/TextController.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.treeview.control

import android.content.Context
import android.view.View
import android.widget.TextView
import org.sqlunet.treeview.model.TreeNode

/**
 * Text controller
 *
 * @param breakExpand whether this controller breaks expansion

 * @author Bogdan Melnychuk on 2/11/15.
 */
class TextController(breakExpand: Boolean) : Controller<Any?>(breakExpand) {

    override fun createNodeView(context: Context, node: TreeNode, value: Any?, minHeight: Int): View {
        val textView = TextView(context)
        if (minHeight > 0) {
            textView.setMinimumHeight(minHeight)
        }
        when (value) {

            is CompositeValue -> {
                textView.text = value.text
                textView.setCompoundDrawablePadding(10)
                textView.setCompoundDrawablesWithIntrinsicBounds(value.icon, 0, 0, 0)
            }

            is CharSequence -> {
                textView.text = value
            }

            else -> {
                throw IllegalArgumentException(value.toString())
            }
        }
        return textView
    }
}
